### DISCUSSION TITLE
How to use SVG object as pattern

### DISCUSSION THREAD

**Discussion by vipin-vidyut** [fabricjs/fabric.js#8683] [General]
Is there any way to use SVG object as pattern without converting it to an Image? 
If SVG used as Image, it's pixel are distorted on zoom.

**Comment by ShaMan123** [comment]
object.toSVG()

**Comment by asturur** [comment]
@vipin-vidyut to do what you describe, if i understand correctly, we should have patterns that behave as native fabricJS objects and that can scale with the zoom.
That is something that always in my head but i never got to code it.

What kind of object are you trying to cover with a pattern?

**Reply by vipin-vidyut** [reply]
@asturur Thanks for reply. I am trying to make floor plan with default shape like rectangle, circle, polygon, etc. with custom pattern like horizontal or vertical or diagonal lines or dotted lines.
If I am use this as SVG object, then it works as expected, to set these SVG as pattern I am using a hack to create Image from SVG and use that Image as pattern, but on zoom Pattern image got distorted.
